src/payment/strategies/googlepay/googlepay-payment-strategy.ts

```diff
@@ -0,0 +1,371 @@
+                ]));
+        }, { methodId }), { queueId: 'widgetInteraction' });
+    }
+
+    private _postForm(paymentData: PaymentSuccessPayload): Promise<Response<any>> {
+        const cardInformation = paymentData.tokenizePayload.details;
+
+        return this._requestSender.post('/checkout.php', {
+            headers: {
+                Accept: 'text/html',
+                'Content-Type': 'application/x-www-form-urlencoded',
+            },
+            body: toFormUrlEncoded({
+                payment_type: paymentData.tokenizePayload.type,
+                nonce: paymentData.tokenizePayload.nonce,
+                provider: this._methodId,
+                action: 'set_external_checkout',
+                card_information: this._getCardInformation(cardInformation),
+            }),
+        });
+    }
+
+    private _getCardInformation(cardInformation: { cardType: string, lastFour: string }) {
+        return {
+            type: cardInformation.cardType,
+            number: cardInformation.lastFour,
+        };
+    }
+
+    private _handleError(error: Error): never {
+        throw new StandardError(error.message);
+    }
+
+    private _getPayment() {
+        return this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(this._methodId))
+            .then(() => {
+                const state = this._store.getState();
+                const paymentMethod = state.paymentMethods.getPaymentMethod(this._methodId);
+
+                if (!paymentMethod) {
+                    throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+                }
+
+                if (!paymentMethod.initializationData.nonce) {
+                    throw new MissingDataError(MissingDataErrorType.MissingPayment);
+                }
+
+                const paymentData = {
+                    method: this._methodId,
+                    nonce: paymentMethod.initializationData.nonce,
+                    cardInformation: paymentMethod.initializationData.card_information,
+                };
+
+                this._paymentMethod = paymentMethod;
+
+                return {
+                    methodId: this._methodId,
+                    paymentData,
+                };
+            });
+    }
+
+    private _updateShippingAndBillingAddress(paymentSucessPayload: PaymentSuccessPayload): Promise<void> {
+        return Promise.all([
+                this._synchronizeShippingAddress(paymentSucessPayload.shippingAddress),
+                this._synchronizeBillingAddress(paymentSucessPayload.billingAddress),
+            ]).then(() => Promise.resolve());
+    }
+
+    private _synchronizeShippingAddress(shippingAddress: GooglePayAddress): Promise<InternalCheckoutSelectors | void> {
+        if (!this._methodId) {
+            throw new RemoteCheckoutSynchronizationError();
+        }
+
+        if (!shippingAddress) {
```

Comment:
You shouldn't need a guard statement here because `shippingAddress` is not an optional parameter.